pkg/apis/node.harvesterhci.io/v1beta1/cloudinit.go

```diff
@@ -0,0 +1,22 @@
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:shortName=nci,scope=Cluster
+
+type CloudInit struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CloudInitSpec `json:"spec"`
```

Comment:
@Vicente-Cheng This is my first "from-scratch" controller, so apologies if this is a naive question.

Since the CloudInit object might not apply to all nodes, would it be OK if the status conditions were "dynamic" based on the node's name?

For example, a condition name would be formatted like `fmt.Sprintf("AppliedTo%s", nodeName)` with a status of "True", "False", or "Unknown" ?

Or is it expected that condition names are static and well-known as part of the controller's behavior?

Reply:
@Vicente-Cheng ping

Reply:
Sorry @connorkuehl, for late reply.
I think you could try to check the `Status.Condition` related article on the k8s.
That would help people to know the status of this CRD.

Reply:
Latest revision consists of a Status member which describes the CloudInit object relevant to each node in the cluster:

```
Status:
  Rollouts:
    Conditions:
      Last Transition Time:  2023-10-24T21:15:02Z
      Message:               
      Reason:                CloudInitApplicable
      Status:                True
      Type:                  Applicable
      Last Transition Time:  2023-10-24T21:09:28Z
      Message:               Local file checksum is the same as the CloudInit checksum
      Reason:                CloudInitChecksumMatch
      Status:                False
      Type:                  OutOfSync
      Last Transition Time:  2023-10-24T21:09:28Z
      Message:               99_ssh.yaml is present under /oem
      Reason:                CloudInitPresentOnDisk
      Status:                True
      Type:                  Present
    Node Name:               harv1
```